src/Lib.P2P/SecureCommunication/Secio1.cs

```diff
@@ -0,0 +1,246 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P.Cryptography;
+using Lib.P2P.Protocols;
+using MultiFormats;
+using Org.BouncyCastle.Security;
+using ProtoBuf;
+using Semver;
+
+namespace Lib.P2P.SecureCommunication
+{
+    /// <summary>
+    ///   Creates a secure connection with a peer.
+    /// </summary>
+    public class Secio1 : IEncryptionProtocol
+    {
+        private static ILog _log = LogManager.GetLogger(typeof(Secio1));
+
+        /// <inheritdoc />
+        public string Name { get; } = "secio";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default)
+        {
+            await EncryptAsync(connection, cancel).ConfigureAwait(false);
+        }
+
+#pragma warning disable VSTHRD103 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)